File: 1002-find-common-characters/1002-find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        int n = words.length;
        int[] count = new int[26];
        Arrays.fill(count, Integer.MAX_VALUE);

        for (int i = 0; i < n; i++) {
            int[] temp = new int[26];
            for (char c : words[i].toCharArray()) {
                temp[c - 'a']++;
            }
            for (int j = 0; j < 26; j++) {
                count[j] = Math.min(count[j], temp[j]);
            }
        }

        List<String> result = new ArrayList<>();
        for (int i = 0; i < 26; i++) {
            while (count[i]-- > 0) {
                result.add(String.valueOf((char)(i + 'a')));
            }
        }

        return result;
    }
}

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        count = [float('inf')] * 26
        n = len(words)
        
        for word in words:
            temp = [0] * 26
            for c in word:
                temp[ord(c) - ord('a')] += 1 
            for i in range(26):
                count[i] = min(count[i], temp[i])

        ans = []
        for i in range(26):
            while count[i] > 0:
                count[i] -= 1
                ans.append(chr(i + ord('a')))
        return ans